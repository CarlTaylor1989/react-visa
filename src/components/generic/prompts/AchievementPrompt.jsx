import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import { Translate } from 'react-redux-i18n';
import achievements from '../../../config/achievements';

import './AchievementPrompt.scss';

class AchievementPrompt extends Component {
  componentDidMount() {
    this.playAudio();
  }

  componentDidUpdate(prevProps) {
    const { achievementId } = this.props;
    if (achievementId && prevProps.achievementId !== achievementId) {
      this.playAudio();
    }
  }

  playAudio() {
    const { setPlayAudio } = this.props;
    setPlayAudio('au2');
  }

  render() {
    const { achievementId, multipleAchievements } = this.props;
    const achievement = achievements.find(ac => ac.id === achievementId);
    const score = achievement ? achievement.score : 0;

    return (
      <div className="achievementPrompt">
        {!multipleAchievements && (
          <Translate
            className="popupTitle"
            value="generic.prompts.achievementTitle"
          />
        )}
        <span className={`achievementIcon dark ${achievementId}`} />
        {multipleAchievements ? (
          <span className="achievementText">
            <Translate
              className="achievementTitle"
              value={`generic.achievements.${achievementId}.title`}
            />
            <Translate
              className="achievementDescription"
              value={`generic.achievements.${achievementId}.description`}
            />
            <div className="score">
              <Translate value="generic.prompts.pointsPrefix" />
              <NumberFormat
                value={score}
                displayType="text"
                thousandSeparator=","
              />
            </div>
          </span>
        ) : (
          <>
            <Translate
              className="achievementTitle"
              value={`generic.achievements.${achievementId}.title`}
            />
            <Translate
              className="achievementDescription"
              value={`generic.achievements.${achievementId}.description`}
            />
            <div className="score">
              <Translate value="generic.prompts.pointsPrefix" />
              <NumberFormat
                value={score}
                displayType="text"
                thousandSeparator=","
              />
            </div>
          </>
        )}
      </div>
    );
  }
}

AchievementPrompt.defaultProps = {
  multipleAchievements: false
};

AchievementPrompt.propTypes = {
  achievementId: PropTypes.string.isRequired,
  multipleAchievements: PropTypes.bool,
  setPlayAudio: PropTypes.func.isRequired
};

export default AchievementPrompt;
