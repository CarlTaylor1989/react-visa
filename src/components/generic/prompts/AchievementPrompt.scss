@import '../../../styles/achievements';
@import '../../../styles/mixins';
@import '../../../styles/variables';

.achievementPrompt {
  text-align: center;

  .achievementIcon {
    @include border-radius($achievement-width);
    display: inline-block;
    margin-bottom: 20px;
  }

  .achievementTitle,
  .achievemenDescription,
  .score {
    display: block;
  }

  .achievementTitle {
    font-size: 24px;
    font-weight: 800;
    margin-bottom: 10px;
  }

  .achievementDescription {
    font-size: 20px;
    margin-bottom: 30px;
  }

  .score {
    color: $colour-yellow;
    font-size: 42px;
    font-weight: 700;
    line-height: 1;
    margin-bottom: 10px;
    margin-top: 30px;
  }
}
