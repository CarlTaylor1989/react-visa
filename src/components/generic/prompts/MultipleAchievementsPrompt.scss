@import '../../../styles/achievements';
@import '../../../styles/mixins';
@import '../../../styles/variables';

.multipleAchievementsPrompt {
  .achievementPrompt {
    display: flex;

    & + .achievementPrompt {
      margin-top: 30px;
    }

    .achievementText {
      flex: 1 0 auto;
      max-width: 100%;
      text-align: left;
    }

    .achievementIcon {
      flex: 0 0 auto;
      margin-right: 30px;
    }

    .achievementDescription {
      display: block;
      margin-bottom: 10px;
    }

    .score {
      margin-top: 0;
    }
  }
}
