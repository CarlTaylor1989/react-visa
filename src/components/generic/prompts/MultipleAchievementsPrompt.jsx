import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';
import AchievementPrompt from './AchievementPrompt';

import './MultipleAchievementsPrompt.scss';

const MultipleAchievements = ({ prompts, setPlayAudio }) => (
  <div className="multipleAchievementsPrompt">
    <Translate
      className="popupTitle multipleAchievementsTitle"
      value="generic.prompts.multipleAchievementsTitle"
    />
    <div className="multipleAchievements">
      {prompts.map(prompt => (
        <AchievementPrompt
          multipleAchievements
          achievementId={prompt.id}
          setPlayAudio={setPlayAudio}
          key={prompt.id}
        />
      ))}
    </div>
  </div>
);

MultipleAchievements.propTypes = {
  prompts: PropTypes.array.isRequired,
  setPlayAudio: PropTypes.func.isRequired,
};


export default MultipleAchievements;
