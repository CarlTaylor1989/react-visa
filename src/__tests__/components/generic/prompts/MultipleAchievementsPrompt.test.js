/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import MultipleAchievementsPrompt from '../../../../components/generic/prompts/MultipleAchievementsPrompt'; // eslint-disable-line max-len
import AchievementPrompt from '../../../../components/generic/prompts/AchievementPrompt';
import {
  ACHIEVEMENT_PROMPT,
} from '../../../../config/constants';

let mockProps;
beforeEach(() => {
  mockProps = {
    prompts: [{
      id: 'ac1',
      type: ACHIEVEMENT_PROMPT
    },
    {
      id: 'ac21',
      type: ACHIEVEMENT_PROMPT
    }],
    setPlayAudio: jest.fn()
  };
});

test('renders an open popup with multiple achievements content', () => {
  const wrapper = shallow(<MultipleAchievementsPrompt {...mockProps} showPopup />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
  expect(wrapper.find('.multipleAchievementsTitle').exists()).toBe(true);
  expect(wrapper.find('.multipleAchievements').exists()).toBe(true);
  expect(wrapper.find(AchievementPrompt)).toHaveLength(mockProps.prompts.length);
});
