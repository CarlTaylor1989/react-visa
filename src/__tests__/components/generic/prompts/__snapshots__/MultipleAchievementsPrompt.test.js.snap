// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an open popup with multiple achievements content 1`] = `
<div
  className="multipleAchievementsPrompt"
>
  <Translate
    className="popupTitle multipleAchievementsTitle"
    tag="span"
    value="generic.prompts.multipleAchievementsTitle"
  />
  <div
    className="multipleAchievements"
  >
    <AchievementPrompt
      achievementId="ac1"
      key="ac1"
      multipleAchievements={true}
      setPlayAudio={[MockFunction]}
    />
    <AchievementPrompt
      achievementId="ac21"
      key="ac21"
      multipleAchievements={true}
      setPlayAudio={[MockFunction]}
    />
  </div>
</div>
`;
